asapdiscovery-docking/asapdiscovery/docking/docking_v2.py

```diff
@@ -71,6 +71,9 @@ def from_pairs(
     def to_design_units(self) -> list[oechem.OEDesignUnit]:
         return [protein_complex.target.to_oedu() for protein_complex in self.complexes]
 
+    def unique_name(self):
+        return f"{self.ligand.compound_name}-{self.ligand.fixed_inchikey}_to_{len(self.complexes)}_targets"
```

Comment:
This also needs to capture the uniqueness of the complexes as well otherwise we will get collisions. You could hash the hash of all the complexes and that should do the trick.

Reply:
What is the unique name being used for?  That is, are you going to be using this for like a file name, or will these just show up in like a pandas data frame?  hashing the collective hashes is probably a good approach to getting a unique identifier, but that isn't very descriptive to a user, so you might want to still have some sort of descriptive string, even if not fully unique.

Reply:
Its for a directory and potentially a cache, so needs to be unique id, but I take your point a bit ugly for a user. For the 1:1 mapping we use: 
```python
    def unique_name(self):
        return f"{self.complex.unique_name()}_{self.ligand.compound_name}-{self.ligand.fixed_inchikey}"
```
Which is semi-human readable but will become very ugly and long with multiple complexes, but perhaps there is no super neat solution.

Reply:
```python 
data = ""
for c in complexes:
    data += c.hash()
hashlib.sha256(data.encode()).hexdigest()
```

I think this should work @apayne97